### ISSUE TITLE
Can't insert with Set command.

### ISSUE THREAD

**Issue by huyphams** [gocassa/gocassa#112]
Hi @hailobackup 

I really like you idea about gocassa. But I have a problem when insert new row to table.

First I create a table by code below:

```

import (
    .....
    "github.com/hailocab/gocassa"
    .....
)
.......
func (r *Resource) GetCassandraTable(name string, e interface{}, p, c []string) gocassa.Table {
    options := gocassa.Options{TableName: name}
    for _, k := range c {
        options = options.AppendClusteringOrder(k, true)
    }
    table := r.Cassandra.Table(name, e, gocassa.Keys{
        PartitionKeys:     p,
        ClusteringColumns: c,
    }).WithOptions(options)
    createIf(table)
    return table
}
.....
```

Code create table:

```
r.UserAlbum = r.GetCassandraTable("user_albums", cassandra.UserAlbum{}, []string{"user"}, []string{"album"})
```

But when create new row:

```
userAlbum := cassandra.UserAlbum{}
userAlbum.User = a.CreatedBy
userAlbum.Album = gocql.TimeUUID()
userAlbum.CreatedAt = a.CreatedAt
userAlbum.Bot = a.Bot
err = r.Resource.UserAlbum.Set(userAlbum).Run()
```

I got error:  **_PRIMARY KEY part user found in SET part**_

But when i changed the source file at: gocassa/table.go line 172: updateOpType -> insertOpType and It works. And I want to know what is the function **_removeFields**_ use for (line 161).

Thanks you.

**Comment by crufter** [comment]
Hey!

Can you gist a file so we can reproduce this?

**Comment by dancannon** [comment]
Hey @huyphams,

Could you please post your data structure so that we can reproduce this issue?

Thanks!

**Comment by dancannon** [comment]
I have attempted to replicate this issue and I seem to get the same error, here is the code I used.

``` go
package main

import (
    "time"

    "github.com/gocql/gocql"

    "github.com/hailocab/gocassa"
)

type UserAlbum struct {
    User      string
    Album     gocql.UUID
    CreatedAt time.Time
    Bot       int
}

func main() {
    keyspace, err := gocassa.ConnectToKeySpace("test", []string{"localhost"}, "", "")
    if err != nil {
        panic(err)
    }

    table := GetCassandraTable(keyspace, "user_albums", UserAlbum{}, []string{"user"}, []string{"album"})

    userAlbum := UserAlbum{}
    userAlbum.User = "daniel.cannon"
    userAlbum.Album = gocql.TimeUUID()
    userAlbum.CreatedAt = time.Now()
    userAlbum.Bot = 1
    err = table.Set(userAlbum).Run()
    if err != nil {
        panic(err)
    }
}

func GetCassandraTable(keyspace gocassa.KeySpace, name string, e interface{}, p, c []string) gocassa.Table {
    options := gocassa.Options{TableName: name}
    for _, k := range c {
        options = options.AppendClusteringOrder(k, true)
    }
    table := keyspace.Table(name, e, gocassa.Keys{
        PartitionKeys:     p,
        ClusteringColumns: c,
    }).WithOptions(options)
    table.CreateIfNotExist()
    return table
}
```

**Comment by dancannon** [comment]
This is now fixed in `v1.1.0`

**Comment by hoffin** [comment]
Just corrected the code above to make it work out the box. The partition key and clustering column passed to the GetCassandraTable call needed capitalising.

```
package main

import (
    "time"

    "github.com/gocql/gocql"

    "github.com/hailocab/gocassa"
)

type UserAlbum struct {
    User      string
    Album     gocql.UUID
    CreatedAt time.Time
    Bot       int
}

func main() {
    keyspace, err := gocassa.ConnectToKeySpace("test", []string{"localhost"}, "", "")
    if err != nil {
        panic(err)
    }

    table := GetCassandraTable(keyspace, "user_albums", UserAlbum{}, []string{"User"}, []string{"Album"})

    userAlbum := UserAlbum{}
    userAlbum.User = "daniel.cannon"
    userAlbum.Album = gocql.TimeUUID()
    userAlbum.CreatedAt = time.Now()
    userAlbum.Bot = 1
    err = table.Set(userAlbum).Run()
    if err != nil {
        panic(err)
    }
}

func GetCassandraTable(keyspace gocassa.KeySpace, name string, e interface{}, p, c []string) gocassa.Table {
    options := gocassa.Options{TableName: name}
    for _, k := range c {
        options = options.AppendClusteringOrder(k, true)
    }
    table := keyspace.Table(name, e, gocassa.Keys{
        PartitionKeys:     p,
        ClusteringColumns: c,
    }).WithOptions(options)
    table.CreateIfNotExist()
    return table
}
```
